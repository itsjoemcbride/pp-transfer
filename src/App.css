@tailwind base;
@tailwind components;
@tailwind utilities;

@layer components {
  .App-header {
    @apply flex min-h-screen flex-col items-center justify-center bg-primary-dark text-[calc(10px+2vmin)] text-white;
  }

  .App-link {
    @apply text-primary-link;
  }

  .App-logo {
    @apply pointer-events-none h-[40vmin] animate-App-logo motion-reduce:animate-none;
  }
}

.balance-box {
  @apply mx-auto mt-36 flex max-w-xl flex-col place-content-center rounded-lg border-4 border-pink-400 bg-white p-4 text-center;
}

body {
  @apply bg-pink-50;
}
