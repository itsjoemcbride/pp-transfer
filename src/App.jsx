import { useState } from "react";
import "./App.css";

function outputMsg(transfer, balance) {
  if (transfer <= balance) {
    setBalance(balance - transfer);
    return `£${transfer} transferred successfully`;
  }
  return `You do not have £${transfer} to transfer`;
}

function App() {
  const [amount, setAmount] = useState("");
  const [transfer, setTransfer] = useState("");
  const [balance, setBalance] = useState(100);

  const handleChange = (event) => setAmount(event.target.value);
  const handleClick = () => {
    setTransfer(amount);
    setAmount("");
  };

  return (
    <>
      <h1>Welcome to PalPay!</h1>
      <h2>Your balance is:</h2>
      <strong>£{balance}</strong>
      <h3>How much would you like to transfer?</h3>
      <input id="amount" value={amount} onChange={handleChange} />
      <button type="button" onClick={handleClick}>
        Send
      </button>
      <p>{outputMsg(transfer, balance)}</p>
    </>
  );
}

export default App;
