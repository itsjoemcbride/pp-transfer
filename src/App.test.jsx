import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import App from "./App";

it("renders without crashing", () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it("updates the balance when we enter some valid input and we click submit", () => {
  render(<App />);

  const balance = screen.getByLabelText(/balance/i);
  const input = screen.getByLabelText(/amount/i);
  const submitBtn = screen.getByRole("button");

  userEvent.type(input, "20");
  userEvent.click(submitBtn);

  expect(balance.textContent).toBe("Â£80");
});

it("updates provides a message when we enter some invalid input and we click submit", () => {
  render(<App />);

  const balance = screen.getByLabelText(/balance/i);
  const input = screen.getByLabelText(/amount/i);
  const submitBtn = screen.getByRole("button");
  const msg = screen.getByLabelText(/message/i);

  userEvent.type(input, "cats");
  userEvent.click(submitBtn);

  expect(balance.textContent).toContain("100");
  expect(msg.textContent).toContain("Insufficient balance");
});
