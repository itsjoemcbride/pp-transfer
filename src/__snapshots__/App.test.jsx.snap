// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="balance-box"
>
  <img
    alt="PalPay Logo"
    className="mx-auto flex h-[80%] w-[80%] justify-center"
    src="PalPay.png"
  />
  <h1>
    Welcome to PalPay!
  </h1>
  <h2>
    Your balance is:
  </h2>
  <strong
    aria-label="balance"
  >
    Â£
    100
  </strong>
  <h3>
    How much would you like to transfer?
  </h3>
  <input
    aria-label="amount"
    className="my-2 mx-auto justify-center rounded-full border py-1 px-2 text-center"
    id="amount"
    onChange={[Function]}
    value=""
  />
  <button
    className="center mx-auto max-w-fit content-center items-center justify-center rounded-full bg-pink-400 px-4 py-2 text-center font-bold text-white hover:bg-pink-600"
    onClick={[Function]}
    type="button"
  >
    Send
  </button>
  <p
    aria-label="message"
    className="pt-2"
  />
</div>
`;
